### PR TITLE
Adding an input filter

### PR DESCRIPTION
**Pull request by tttp**
Thanks to the suggestion on github, added throttling. It turns out that the result is smooth enough by simply waiting until the 200ms, rather than complicating and offering the option to draw separately slow and faster graphs.

The spec needs better test, but I'd need some help, I'm a bit lost

I'd love if someone could comment on the feature itself, I'll figure out what makes sense to test soon I think

### CODE REVIEW

File: src/filter-input.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+      _chart.root().html(_html());
+
+      _chart.root().selectAll("input").on("input", function() {
+        _filter(_normalize(this.value));
+        _throttle();
+      });
+      return _chart;
+    };
+
+    var throttleTimer;
```
Comment (gordonwoodhull) [inline src/filter-input.js]: the convention is to use underscore before private variables, and put them at the top of the "class"

**Comment by gordonwoodhull** [comment]
I've rebased this to a branch: https://github.com/dc-js/dc.js/tree/input-filter

We just need to fix those tests and this can go in.

**Comment by rrameshkumar76** [comment]
input filter looks to be a great feature to have in dc library.

**Comment by gordonwoodhull** [comment]
Hi @rrameshkumar76, I hope to merge this soon. 

If you want to help, you could try this PR, either by checking out the branch I linked above, or simply by pasting the code into your copy of dc.js.

Then please leave a review here in the ticket: if it worked for your purpose, if you think anything is missing, etc. Thanks!

**Comment by cmorgan** [comment]
this works for me. Is there a way to hook into changes to the inputFilter so I can redraw a datatables.js table? For other widgets I am hooking into dc.chartRegistry.list()[i].on("filtered") and the redrawing the table, is there an equivalent hook for a inputFilter?

**Comment by kum-deepak** [comment]
@tttp Thanks for your excellent work. I am currently working on upgrading dc to use D3v4 and assisting on preparing next major release of dc (version 3). You can track the progress in 3.0 branch. @gordonwoodhull has planned this PR to be part of this release. I will be working on upgrading this PR for D3v4 and readying for inclusion in dc.

Will really appreciate if you are around and are able to help :)

**Comment by kum-deepak** [comment]
Please check #1387

**Comment by tttp** [comment]
Obsoleted by #1387

**Pull request closed without merging**